rts/motoko-rts/src/types.rs

```diff
@@ -256,63 +257,120 @@ impl Value {
     }
 
     /// Assumes that the value is a pointer and returns the pointer value. In debug mode panics if
-    /// the value is not a pointer.
+    /// the value is not a pointer or if it points to a stale object that has been forwarded.
     pub fn get_ptr(self) -> usize {
         debug_assert!(self.get().is_ptr());
         unskew(self.0 as usize)
     }
 
-    /// Get the object tag. In debug mode panics if the value is not a pointer.
+    /// Check that the forwarding pointer is valid.
+    #[inline]
+    pub unsafe fn check_forwarding_pointer(self) {
+        debug_assert!(
+            self.forward().get_ptr() == self.get_ptr()
+                || self.forward().forward().get_ptr() == self.forward().get_ptr()
```

Comment:
In the current PR, an even stronger invariant `self.forward().get_ptr() == self.get_ptr()` should hold, right?

Reply:
Indeed. It is a preparation for the third PR (probably a left-over when I split the PRs). I will update this for this PR.